main.go

```diff
@@ -136,7 +138,7 @@ func main() {
 	// Controllers who register after manager starts will start directly.
 	go setupControllers(mgr, certsReady)
 
-	setupHealthzAndReadyzCheck(mgr)
+	setupHealthzAndReadyzCheck(mgr, certsReady)
```

Comment:
This adds a second consumer to the same `certsReady` channel that `setupControllers()` function (line 137) is already using to receive a message indicating certs are ready.  If the producer in the function `cert.CertsManager(mgr, certsReady)` above (line 121) only sends one message in the channel, then only one of the consumers will receive the signal certs are ready.

Can you double check the expected behavior of `cert.CertsManager(mgr, certsReady)` and verify have 2 consumers like this will work?

Reply:
It's not quite like that. After creating the certificates, the `certsReady` channel will be closed. And when we close the channel, all subscribers get a message.

Reply:
Ah ok, got it.